### ISSUE TITLE
Why does my loss decrease unstabley when I train the net with 500 patches?  Because the data is too little? 

### ISSUE THREAD

**Issue by yuyadanyadan** [nagadomi/waifu2x#146]
I1019 16:56:37.955636  8822 solver.cpp:305] Learning Rate Policy: waifu2x
I1019 16:56:37.956248  8822 solver.cpp:362] Iteration 0, Testing net (#0)
I1019 16:57:55.966459  8822 solver.cpp:429]     Test net output #0: loss = 0.0290823 (* 1 = 0.0290823 loss)
I1019 16:58:00.868696  8822 solver.cpp:253] Iteration 0, loss = 0.0311817
I1019 16:58:00.868749  8822 solver.cpp:269]     Train net output #0: loss = 0.0311817 (* 1 = 0.0311817 loss)
I1019 16:58:00.868759  8822 sgd_solver.cpp:116] Iteration 0, lr = 0.000249999
I1019 17:04:51.740345  8822 solver.cpp:253] Iteration 100, loss = 0.040274
I1019 17:04:51.740445  8822 solver.cpp:269]     Train net output #0: loss = 0.040274 (* 1 = 0.040274 loss)
I1019 17:04:51.740456  8822 sgd_solver.cpp:116] Iteration 100, lr = 0.000249924
I1019 17:12:32.962102  8822 solver.cpp:253] Iteration 200, loss = 0.0449925
I1019 17:12:33.355516  8822 solver.cpp:269]     Train net output #0: loss = 0.0449925 (* 1 = 0.0449925 loss)
I1019 17:12:33.355578  8822 sgd_solver.cpp:116] Iteration 200, lr = 0.000249849
I1019 17:19:23.685695  8822 solver.cpp:253] Iteration 300, loss = 0.0448228
I1019 17:19:23.849676  8822 solver.cpp:269]     Train net output #0: loss = 0.0448229 (* 1 = 0.0448229 loss)
I1019 17:19:23.849730  8822 sgd_solver.cpp:116] Iteration 300, lr = 0.000249774
I1019 17:26:06.037768  8822 solver.cpp:253] Iteration 400, loss = 0.0264396
I1019 17:26:06.037914  8822 solver.cpp:269]     Train net output #0: loss = 0.0264396 (* 1 = 0.0264396 loss)
I1019 17:26:06.037925  8822 sgd_solver.cpp:116] Iteration 400, lr = 0.0002497
I1019 17:32:44.985211  8822 solver.cpp:479] Snapshotting to binary proto file /home/yuyadan/caffe/examples/waifu2x_trainning_cpp/waifu2x_train_iter_500.caffemodel
I1019 17:32:45.585852  8822 sgd_solver.cpp:290] Snapshotting solver state to binary proto file /home/yuyadan/caffe/examples/waifu2x_trainning_cpp/waifu2x_train_iter_500.solverstate
I1019 17:32:49.843364  8822 solver.cpp:253] Iteration 500, loss = 0.0255457
I1019 17:32:49.843487  8822 solver.cpp:269]     Train net output #0: loss = 0.0255457 (* 1 = 0.0255457 loss)
I1019 17:32:49.843502  8822 sgd_solver.cpp:116] Iteration 500, lr = 0.000249625
I1019 17:39:36.430296  8822 solver.cpp:253] Iteration 600, loss = 0.037344
I1019 17:39:36.430485  8822 solver.cpp:269]     Train net output #0: loss = 0.0373441 (* 1 = 0.0373441 loss)
I1019 17:39:36.430498  8822 sgd_solver.cpp:116] Iteration 600, lr = 0.00024955
I1019 17:46:21.068217  8822 solver.cpp:253] Iteration 700, loss = 0.0357204
I1019 17:46:21.068420  8822 solver.cpp:269]     Train net output #0: loss = 0.0357205 (* 1 = 0.0357205 loss)
I1019 17:46:21.068434  8822 sgd_solver.cpp:116] Iteration 700, lr = 0.000249475
I1019 17:53:04.696640  8822 solver.cpp:253] Iteration 800, loss = 0.0162388
I1019 17:53:04.696722  8822 solver.cpp:269]     Train net output #0: loss = 0.0162389 (* 1 = 0.0162389 loss)
I1019 17:53:04.696733  8822 sgd_solver.cpp:116] Iteration 800, lr = 0.000249401
I1019 17:59:45.030585  8822 solver.cpp:253] Iteration 900, loss = 0.0165265
I1019 17:59:45.030730  8822 solver.cpp:269]     Train net output #0: loss = 0.0165266 (* 1 = 0.0165266 loss)
I1019 17:59:45.030741  8822 sgd_solver.cpp:116] Iteration 900, lr = 0.000249326
I1019 18:06:30.505097  8822 solver.cpp:479] Snapshotting to binary proto file /home/yuyadan/caffe/examples/waifu2x_trainning_cpp/waifu2x_train_iter_1000.caffemodel
I1019 18:06:30.517384  8822 sgd_solver.cpp:290] Snapshotting solver state to binary proto file /home/yuyadan/caffe/examples/waifu2x_trainning_cpp/waifu2x_train_iter_1000.solverstate
I1019 18:06:30.522547  8822 solver.cpp:362] Iteration 1000, Testing net (#0)
I1019 18:07:46.129395  8822 solver.cpp:429]     Test net output #0: loss = 0.0365902 (* 1 = 0.0365902 loss)
I1019 18:07:50.044318  8822 solver.cpp:253] Iteration 1000, loss = 0.032317
I1019 18:07:50.044371  8822 solver.cpp:269]     Train net output #0: loss = 0.0323171 (* 1 = 0.0323171 loss)
I1019 18:07:50.044381  8822 sgd_solver.cpp:116] Iteration 1000, lr = 0.000249252
I1019 18:14:30.554181  8822 solver.cpp:253] Iteration 1100, loss = 0.0319682
I1019 18:14:30.554280  8822 solver.cpp:269]     Train net output #0: loss = 0.0319683 (* 1 = 0.0319683 loss)
I1019 18:14:30.554291  8822 sgd_solver.cpp:116] Iteration 1100, lr = 0.000249177
I1019 18:21:23.975481  8822 solver.cpp:253] Iteration 1200, loss = 2.0083e-05
I1019 18:21:23.975605  8822 solver.cpp:269]     Train net output #0: loss = 2.0218e-05 (* 1 = 2.0218e-05 loss)
I1019 18:21:23.975615  8822 sgd_solver.cpp:116] Iteration 1200, lr = 0.000249103
I1019 18:28:05.186214  8822 solver.cpp:253] Iteration 1300, loss = 1.34425e-05
I1019 18:28:05.186306  8822 solver.cpp:269]     Train net output #0: loss = 1.35886e-05 (* 1 = 1.35886e-05 loss)
I1019 18:28:05.186317  8822 sgd_solver.cpp:116] Iteration 1300, lr = 0.000249028
I1019 18:35:01.660615  8822 solver.cpp:253] Iteration 1400, loss = 0.0427325
I1019 18:35:01.660747  8822 solver.cpp:269]     Train net output #0: loss = 0.0427326 (* 1 = 0.0427326 loss)
I1019 18:35:01.660758  8822 sgd_solver.cpp:116] Iteration 1400, lr = 0.000248954
I1019 18:41:42.918938  8822 solver.cpp:479] Snapshotting to binary proto file /home/yuyadan/caffe/examples/waifu2x_trainning_cpp/waifu2x_train_iter_1500.caffemodel
I1019 18:41:42.925552  8822 sgd_solver.cpp:290] Snapshotting solver state to binary proto file /home/yuyadan/caffe/examples/waifu2x_trainning_cpp/waifu2x_train_iter_1500.solverstate
I1019 18:41:47.056991  8822 solver.cpp:253] Iteration 1500, loss = 0.0451393
I1019 18:41:47.057047  8822 solver.cpp:269]     Train net output #0: loss = 0.0451395 (* 1 = 0.0451395 loss)
I1019 18:41:47.057057  8822 sgd_solver.cpp:116] Iteration 1500, lr = 0.000248879
I1019 18:48:31.680407  8822 solver.cpp:253] Iteration 1600, loss = 0.0348224
I1019 18:48:31.680615  8822 solver.cpp:269]     Train net output #0: loss = 0.0348226 (* 1 = 0.0348226 loss)
I1019 18:48:31.680629  8822 sgd_solver.cpp:116] Iteration 1600, lr = 0.000248805
I1019 18:55:14.063782  8822 solver.cpp:253] Iteration 1700, loss = 0.0411839
I1019 18:55:14.063937  8822 solver.cpp:269]     Train net output #0: loss = 0.0411841 (* 1 = 0.0411841 loss)
I1019 18:55:14.063948  8822 sgd_solver.cpp:116] Iteration 1700, lr = 0.000248731
I1019 19:02:05.532102  8822 solver.cpp:253] Iteration 1800, loss = 0.0407743
I1019 19:02:05.532209  8822 solver.cpp:269]     Train net output #0: loss = 0.0407745 (* 1 = 0.0407745 loss)
I1019 19:02:05.532230  8822 sgd_solver.cpp:116] Iteration 1800, lr = 0.000248657

**Comment by nagadomi** [comment]
It seems using caffe. I am not familiar with caffe and I don't know about your code.
Maybe code has a bug or some hyperparameter is inappropriate. I don't have advice for this.
